crates/solidity/outputs/spec/build.rs

```diff
@@ -1,12 +1,88 @@
+use std::path::Path;
+
 use anyhow::Result;
-use codegen_schema::types::LanguageDefinition;
+use codegen_ebnf::{EbnfModel, EbnfWriter};
+use codegen_language_definition::model::Language;
 use codegen_spec::SpecGeneratorExtensions;
 use infra_utils::cargo::CargoWorkspace;
-use solidity_language::SolidityLanguageExtensions;
+use infra_utils::codegen::{Codegen, CodegenWriteOnly};
+use solidity_language::{SolidityDefinition, SolidityLanguageExtensions};
 
 fn main() -> Result<()> {
-    let language = LanguageDefinition::load_solidity()?;
+    let mut codegen = Codegen::write_only()?;
     let output_dir = CargoWorkspace::locate_source_crate("solidity_spec")?.join("generated");
 
-    language.generate_spec(&output_dir)
+    // TODO: combine both ebnf and spec codegen into 'codegen_spec' crate:
+    let language_v0 = codegen_schema::types::LanguageDefinition::load_solidity()?;
+    language_v0.generate_spec(&mut codegen, &output_dir)?;
+
+    let language_v2 = SolidityDefinition::create();
+    generate_language_ebnf(&mut codegen, &language_v2, &output_dir)?;
+
+    Ok(())
+}
+
+fn generate_language_ebnf(
+    codegen: &mut CodegenWriteOnly,
+    language: &Language,
+    output_dir: &Path,
+) -> Result<()> {
+    let mut writer = SimpleEbnfWriter {
+        buffer: String::new(),
+    };
+    let ebnf_model = EbnfModel::build(language);
+
+    for (section_index, section) in language.sections.iter().enumerate() {
+        writer.buffer.push_str(&format!(
+            "(*\n * {section_index}. {section_title}:\n *)\n\n",
+            section_index = section_index + 1,
+            section_title = section.title,
+        ));
```

Comment:
Same as below; it'd be nice to only go through the `EbnfWriter` here (and I don't think multi-line gives us much, if we're worried about abstraction leakage and having to include ` *` prefix explicitly)

Reply:
Sounds good. Will do.

Reply:
This writer is solely for writing `.ebnf` files. The abstraction is here to allow us to write other formats, like HTML (syntax higlighting in the rendered spec).